src/reducers/alerts.js

```diff
@@ -126,10 +153,30 @@ const showExtensionAlert = function (data) {
     };
 };
 
+/**
+ * Function to dispatch showing an alert, with optional
+ * timeout to make it close/go away.
+ *
+ * @param {object} alertId - the ID of the alert
+ * @return {null} - do not return a value
+ */
+const showAlertWithTimeout = alertId => (dispatch => {
```

Comment:
Redux does not like this async stuff being in a thunk function!

Giving me: 
```
    Actions must be plain objects. Use custom middleware for async actions.
```

Any ideas?

Reply:
It seems like the thunk middleware is configured correctly. Is that happening in development as well as in the tests?

Could you see if you could do this with a handler instead of building it into the action?

Reply:
OK, I revised my approach along the lines you suggested, and removed thunk.